packages/core/src/cache/cache.ts

```diff
@@ -1080,6 +1080,19 @@ class Cache {
     this._geometryCacheSize -= decrement;
   };
 
+  /**
+   * Find the image that has the referenced ImageId
+   */
+  public getImageByReferencedImageId = (
```

Comment:
Probably you should maintain a map for referenced image id to cached image.  Otherwise this will get really slow.